### DISCUSSION TITLE
No limit strings

### DISCUSSION THREAD

**Discussion by raorsa** [Laravel-Backpack/community-forum#747] [Ideas]
Have the possibility of not limiting the value of text columns, setting the LIMIT value to 0

text.blade.php
line 19
        $column['text'] = $column['prefix'].($column['limit']>0?Str::limit($column['value'], $column['limit'], '…'):$column['value']).$column['suffix'];

**Comment by pxpm** [comment]
Hey @raorsa thanks for the suggestion. 🙏 

Although the solution doesn't seem that trivial to me. As a developer, I can use the limit with negative values to cap a string. 

Imagine that you have something like `pdf-document-1246, another-document-6421, one-more-1234`, you know that you want to "remove" the last 5 (`-xxxx`) so you could just do `limit => -5` and would get: `pdf-document..., another-document..., one-more...`. With your proposed solution this would become impossible. 

I would rather allow the `limit` to be `false|int` so that you as developer could intentionally disable the limit feature, but I think an absurd `limit` number works just as well as the false.

You can set the `limit` to a very big number, like `limit => 999999999`. I am pretty sure if you are trying to display something that bigger in the table, anyone using it will have a very hard time to navigate it, so anything smaller but that you don't know if it's 100, or 3000 should be covered and wont it the limit.

I am closing this as I don't think there is any need to keep this open, we are happy with the limit "workaround", but feel free to continue the conversation if you think I am wrong here. I have been wrong many times before 👍 

Cheers

**Reply by raorsa** [reply]
Ok, thank you very much for the advice on the negative number, I think the approximation of
$column['limit']==0
It makes more sense to cover all scenarios, using a meaningless number like 0 for character limitation.

**Reply by pxpm** [reply]
Indeed, making it `=== 0` would work slightly better but would still misleading to the developers and a potential breaking change for very to no benefits over an absurdly big number. 

I think the only reasonable solution that would leave the `Str::limit()` do it's job either with `-1, 0 or 1` is to introduce the `boolean`. 
So a `limit => false` won't apply any limit. Anything around integers would require us to change the `Str::limit()` API, and there may be use cases where people are passing `Str::limit(0)` and showing only something in the `prefix` for example, I don't know 🤷 

In a future general refactor (probably next major version), we can think on a better API around limit and built it from ground up with the possibility to be disabled.
Changing now 8~10 fields that use the limit with a possible breaking change on the back is not worth it at the moment. 

Hope it makes sense. 

Cheers
